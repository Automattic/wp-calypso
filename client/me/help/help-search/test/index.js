/**
 * @format
 * @jest-environment jsdom
 */

/**
 * External dependencies
 */
import React from 'react';
import { identity } from 'lodash';
import { shallow } from 'enzyme';

/**
 * Internal dependencies
 */

import { HelpSearch } from '../';

const defaultProps = {
	translate: identity,
};

const helpLinks = {
	wordpress_forum_links_localized: [
		{
			description: 'bacon',
			link: 'apple',
			title: 'pancake',
		},
	],
	wordpress_forum_links: [
		{
			description: 'walnut',
			link: 'apricot',
			title: 'cream',
		},
	],
};

const searchQuery = 'maplesyrup';

describe( 'HelpSearch', () => {
	test( 'should render ', () => {
		const wrapper = shallow( <HelpSearch { ...defaultProps } /> );
		expect( wrapper ).toMatchSnapshot();
	} );

	test( 'should prioritize and pass localized forum search results if available ', () => {
		const wrapper = shallow( <HelpSearch { ...defaultProps } helpLinks={ helpLinks } /> );
		wrapper.setState( { searchQuery } );
		wrapper.update();
		const HelpResultsElements = wrapper.find( 'HelpResults' );
		expect( HelpResultsElements ).toHaveLength( 3 );
		expect( HelpResultsElements.at( 1 ).props().helpLinks ).toEqual(
			helpLinks.wordpress_forum_links_localized
		);
	} );

	test( 'should display `en` forum search results localized not available', () => {
		const wrapper = shallow(
			<HelpSearch
				{ ...defaultProps }
				helpLinks={ { wordpress_forum_links: helpLinks.wordpress_forum_links } }
			/>
		);
		wrapper.setState( { searchQuery } );
		wrapper.update();
		const HelpResultsElements = wrapper.find( 'HelpResults' );
		expect( HelpResultsElements ).toHaveLength( 3 );
		expect( HelpResultsElements.at( 1 ).props().helpLinks ).toEqual(
			helpLinks.wordpress_forum_links
		);
	} );
} );
