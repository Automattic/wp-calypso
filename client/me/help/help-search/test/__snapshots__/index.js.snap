// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpSearch should render  1`] = `
<div
  className="help-search"
>
  <SearchCard
    analyticsGroup="Help"
    delaySearch={true}
    onSearch={[Function]}
    placeholder="How can we help?"
  />
</div>
`;
